File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.java
import java.util.Scanner;
import java.util.HashSet;
import java.util.ArrayList;
import java.util.Set;
import java.util.Collections;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        sc.nextLine(); // 버퍼 비우기

        Set<String> log = new HashSet<>();

        for (int i = 0; i < n; i++) {
            String[] input = sc.nextLine().split(" ");
            String name = input[0];
            String status = input[1];

            if (status.equals("enter")) {
                log.add(name);
            } else {
                log.remove(name);
            }
        }

        // Set -> ArrayList
        ArrayList<String> result = new ArrayList<>(log);
        Collections.sort(result, Collections.reverseOrder());

        for (String name : result) {
            System.out.println(name);
        }

        sc.close();
    }
}

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.py
import sys
input = sys.stdin.readline

n = int(input())
log = set()

for _ in range(n):
    name, status = input().split()

    if status == 'enter':
        log.add(name)
    else:
        log.remove(name)


for name in sorted(log, reverse=True):
    print(name)